ledger/ledgercore/statedelta.go

```diff
@@ -193,7 +199,7 @@ func (sd *StateDelta) OptimizeAllocatedMemory(proto config.ConsensusParams) {
 	}
 
 	// Creatables takes up 100 bytes per entry, and is saved for 320 rounds
-	if uint64(sd.initialTransactionsCount-len(sd.Creatables))*creatablesEntrySize*proto.MaxBalLookback > stateDeltaTargetOptimizationThreshold {
+	if uint64(len(sd.Creatables))*creatablesEntrySize*proto.MaxBalLookback > stateDeltaTargetOptimizationThreshold {
```

Comment:
Why even bother rewriting this map if you are not supplying a hint when creating it anymore?